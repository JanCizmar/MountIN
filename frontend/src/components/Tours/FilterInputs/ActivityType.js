import React from 'react';
import {FormGroup} from "react-bootstrap";
import Select from "react-select";
import PropTypes from 'prop-types';
import {ActivityType as Types} from '../../../constantLists/ActivityType';

const typeOptions = Object.keys(Types.names).map((key) => {
    return {label: Types.names[key], value: key}
});

export const ActivityType = (props) =>
    <FormGroup>
        <Select isMulti={true} options={typeOptions}
                onChange={props.onChange}
                value={props.value}

        />
    </FormGroup>;

ActivityType.propTypes = {
    onChange: PropTypes.func.isRequired,
    value: PropTypes.array.isRequired
};