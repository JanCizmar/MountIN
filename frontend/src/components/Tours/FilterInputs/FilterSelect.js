import React from 'react';
import Select from "react-select";

export const FilterSelect = (props) => {
    let onChange = (vals) =>
        props.onChange(vals.map((val) => val.value)); // return just values, not labels

    let value = props.value.map((val) => props.options.find(option => option.value === val)); //convert the value to expected

    return <Select {...props}
                   onChange={onChange}
                   value={value}
    />
};