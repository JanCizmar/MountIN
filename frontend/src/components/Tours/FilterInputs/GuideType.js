import React from 'react';
import {FormGroup} from "react-bootstrap";
import Select from "react-select";
import PropTypes from 'prop-types';
import {GuideType as Types} from '../../../constantLists/GuideType';

const typeOptions = Object.keys(Types.names).map((key) => {
    return {label: Types.names[key], value: key}
});

export const GuideType = (props) =>
    <FormGroup>
        <Select isMulti={true} options={typeOptions}
                onChange={props.onChange}
                value={props.value}/>
    </FormGroup>;

GuideType.propTypes = {
    onChange: PropTypes.func.isRequired,
    value: PropTypes.array.isRequired
};