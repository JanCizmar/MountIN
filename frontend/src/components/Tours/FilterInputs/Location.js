import React from 'react';
import {FormControl, FormGroup} from "react-bootstrap";
import PlacesAutocomplete, {geocodeByAddress, getLatLng} from 'react-places-autocomplete'
import {compose, withHandlers} from "recompose";
import PropTypes from 'prop-types';

export const Location = compose(
    withHandlers({
        onSelect: props => address => {
            props.onValueChange(address);
            if ((props.onLatLngChange) !== undefined) {
                props.onLatLngLoadingStarts && props.onLatLngLoadingStarts();
                geocodeByAddress(address).then(results => {
                    getLatLng(results[0]).then(res => {
                      props.onLatLngChange(res);
                      props.onLatLngLoadingFinished && props.onLatLngLoadingFinished();
                    });
                });
            }
        }
    })
)((props) =>
    <FormGroup className="location"
        controlId="location">
        <PlacesAutocomplete onChange={props.onValueChange} value={props.value} onSelect={props.onSelect}>
            {({getInputProps, suggestions, getSuggestionItemProps}) => (
                <div>
                    <FormControl
                        {...getInputProps({
                            placeholder: 'Search Places ...',
                            className: 'location-search-input'
                        })}
                    />
                    <div className={suggestions.length && 'autocomplete-dropdown-container'}>
                        {suggestions.map(suggestion => {
                            const className = suggestion.active ? 'suggestion-item active' : 'suggestion-item';
                            return (
                                <div {...getSuggestionItemProps(suggestion, {className})}>
                                    <span>{suggestion.description}</span>
                                </div>
                            )
                        })}
                    </div>
                </div>
            )}
        </PlacesAutocomplete>
    </FormGroup>
);

Location.propTypes = {
    onLatLngChange: PropTypes.func,
    onValueChange: PropTypes.func.isRequired,
    onLatLngLoadingStarts: PropTypes.func,
    onLatLngLoadingFinished: PropTypes.func,
    value: PropTypes.string.isRequired
};