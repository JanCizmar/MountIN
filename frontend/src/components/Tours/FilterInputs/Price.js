import React from 'react';
import {FormGroup} from "react-bootstrap";
import PropTypes from 'prop-types';
import Range from 'rc-slider/lib/Range';

export const Price = (props) =>
    <div className="price-wrapper">
        <FormGroup className="price">
            <div className="value min">{props.value[0] ? props.value[0] : 'Free'}</div>
            <Range min={0} max={500} value={props.value} onChange={props.onChange}/>
            <div className="value max">{props.value[1]}&nbsp;â‚¬</div>
        </FormGroup>
    </div>;

Price.propTypes = {
    onChange: PropTypes.func.isRequired,
    value: PropTypes.array.isRequired
};