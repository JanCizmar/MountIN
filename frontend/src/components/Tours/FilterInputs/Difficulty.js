import React from 'react';
import {FormGroup} from "react-bootstrap";
import PropTypes from 'prop-types';
import {Difficulities as Types} from '../../../constantLists/Difficulities';
import {FilterSelect} from "./FilterSelect";

const typeOptions = Object.keys(Types.names).map((key) => {
    return {label: Types.names[key], value: key}
});

export const Difficulty = (props) =>
    <FormGroup>
        <FilterSelect isMulti={true} options={typeOptions}
                      onChange={props.onChange}
                      value={props.value}
                      placeholder='Difficulty'
        />
    </FormGroup>;

Difficulty.propTypes = {
    onChange: PropTypes.func.isRequired,
    value: PropTypes.array.isRequired
};