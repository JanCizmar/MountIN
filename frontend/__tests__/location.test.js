import React from 'react';
import renderer from 'react-test-renderer';
import {Location} from "../src/components/Tours/FilterInputs/Location";
//Google maps mock
window.google = {
    maps: {
        places: {
            PlacesServiceStatus: {
                OK: true
            },
            AutocompleteService: class {
            }
        }
    }
};

it('renders correctly', () => {
    const tree = renderer
        .create(<Location value={{}} onValueChange={() => {
        }}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});