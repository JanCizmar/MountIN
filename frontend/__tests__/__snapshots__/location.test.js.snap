// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="location form-group"
>
  <div>
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      className="location-search-input form-control"
      disabled={false}
      id="location"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Location"
      role="combobox"
      type="text"
      value=""
    />
    <div
      className={0}
    />
  </div>
</div>
`;
